src/bokeh/models/formatters.py

```diff
@@ -699,6 +699,55 @@ def RELATIVE_DATETIME_CONTEXT() -> DatetimeTickFormatter:
         years = "",
     )
 
+def DYNAMIC_DATETIME_FORMATTER() -> DatetimeTickFormatter:
+    return DatetimeTickFormatter(**_DYNAMIC_DATETIME_KWARGS())
+
+def _DYNAMIC_DATETIME_KWARGS() -> dict:
+    return {
+        "microseconds": "%fus",
+        "milliseconds": "%3Nms",
```

Comment:
I'm not sure why this code was structured this way, but it prevents type checking (if models provided useful type information), auto-completion/intellisense and makes refactoring or other automated changes harder.

Reply:
You mean why the kwargs are in a separate function (I can adapt this as it is just a relict from handling InstanceDefault) or why the default values are managed here in the first place?

Reply:
I expected this:
```suggestion
def DYNAMIC_DATETIME_FORMATTER() -> DatetimeTickFormatter:
    return DatetimeTickFormatter(
        microseconds: "%fus",
        milliseconds: "%3Nms",
```
and so on.

Reply:
Fixed now.